artemis-server/src/main/resources/schema/artemis-configuration.xsd

```diff
@@ -675,6 +675,82 @@
             </xsd:annotation>
          </xsd:element>
 
+         <xsd:element name="journal-retention" maxOccurs="1" minOccurs="0">
+            <xsd:annotation>
+               <xsd:documentation>
+                  Configuration about the journal retention configuration.
+               </xsd:documentation>
+            </xsd:annotation>
+            <xsd:complexType>
+               <xsd:attribute name="unit" use="required">
+                  <xsd:annotation>
+                     <xsd:documentation>
+                        This configures the period type to use on limit. By default it is DAYS.
+                     </xsd:documentation>
+                  </xsd:annotation>
+                  <xsd:simpleType>
+                     <xsd:restriction base="xsd:string">
+                        <xsd:enumeration value="DAYS"/>
+                        <xsd:enumeration value="HOURS"/>
+                        <xsd:enumeration value="SECONDS"/>
+                     </xsd:restriction>
+                  </xsd:simpleType>
+               </xsd:attribute>
+               <xsd:attribute name="directory" type="xsd:string" use="required">
+                  <xsd:annotation>
+                     <xsd:documentation>
+                        The location on your file system where the journal-retention will take place.
+                     </xsd:documentation>
+                  </xsd:annotation>
+               </xsd:attribute>
+               <xsd:attribute name="period" type="xsd:integer" use="required">
+                  <xsd:annotation>
+                     <xsd:documentation>
+                        The amount of time used to keep files.
```

Comment:
What if someone wants to limit just by size?

Reply:
maybuscan use xsd assert to handle more complex checks that at least size or time is set and if time that period is set

Reply:
I think we should require a period configured. make it large if you wish (365 days e.g)

Reply:
I will allow unlimited.. and log a warn if no value is provided.

Reply:
I will actually require one filled.. either period or storage and throw an error if not passed.